src/libawkward/array/RecordArray.cpp

```diff
@@ -360,6 +360,11 @@ namespace awkward {
     return util::keys(recordlookup_, numfields());
   }
 
+  int64_t RecordArray::list_depth() const {
+    // FIXME: find a specific one?
+    return contents_[0].get()->list_depth() + (int64_t)1;
```

Comment:
This is why we have `minmax_depth`; both `RecordArray` (product types) and `UnionArray` (sum types) can have ambiguous list depths. One way of handling this would be to say that operations like `flatten` can't deal with ambiguous lengths. Another way is to convert negative `axis` values to _different_ positive values, depending on the branch of the `RecordArray`/`UnionArray`.

Reply:
done.